src/common.jl

```diff
@@ -59,7 +59,7 @@ function solve{uType,tType,isinplace,F,Method,LinearSolver}(
         method_code = CV_FUNCTIONAL
     end
 
-    mem = CVodeCreate(alg_code, method_code)
+    mem = Handle(CVodeCreate(alg_code, method_code))
```

Comment:
ATM `Handle{}` doesn't check If it's being passed the null pointer.
I would throw an `ArgumentError` in the `Handle` constructor in this case.
Here (and for other XXXCreate) I would leave the old `mem == C_NULL` check (as it generates the sensible error message) and create a handle only if `mem` is non null.

An alternative would be to allow null handles, but then `release_handle()` should have a null check. I would also add `isnull(Handle)` method to avoid comparing handles with integers.